pkg/protocol/engine/blocks/block.go

```diff
@@ -253,20 +258,33 @@ func (b *Block) IsMissing() (isMissing bool) {
 	return b.missing
 }
 
+// Solid returns a reactive variable that is true if the Block is solid (the entire causal history is known).
+func (b *Block) Solid() (solid reactive.Variable[bool]) {
+	b.mutex.RLock()
```

Comment:
why do we need locks here to protect these variables?

Reply:
We don't on the reactive variables :+1: